packages/docmanager/src/tokens.ts

```diff
@@ -36,6 +36,11 @@ export interface IDocumentManager extends IDisposable {
    */
   readonly activateRequested: ISignal<this, string>;
 
+  /**
+   * A signal emitted when an option has changed.
+   */
+  readonly optionChanged: ISignal<this, any>;
```

Comment:
Ideally we wouldn't need this signal either, and the logic on whether `nameFileOnSave` is enabled could be kept at the plugin level.

Reply:
We can do a second pass in a separate change to check that.